packages/basic-crawler/src/internals/error-tracker.ts

```diff
@@ -0,0 +1,345 @@
+
+                if (streak > maxStreak) {
+                    maxStreak = streak;
+                    aStreakIndex = aIndex;
+                    bStreakIndex = bIndex;
+                }
+            } while (bIndex !== -1);
+        }
+
+        return {
+            maxStreak,
+            aStreakIndex,
+            bStreakIndex,
+        };
+    };
+
+    const calculatePlaceholder = (a: string[], b: string[]) => {
+        const { maxStreak, aStreakIndex, bStreakIndex } = findBiggestWordIntersection(a, b);
+
+        if (maxStreak === 0) {
+            return ['_'];
+        }
+
+        const leftA = a.slice(0, aStreakIndex);
+        const leftB = b.slice(0, bStreakIndex);
+        const rightA = a.slice(aStreakIndex + maxStreak);
+        const rightB = b.slice(bStreakIndex + maxStreak);
+
+        const output: string[] = [];
+
+        if (leftA.length !== 0 || leftB.length !== 0) {
+            output.push(...calculatePlaceholder(leftA, leftB));
+        }
+
+        output.push(...a.slice(aStreakIndex, aStreakIndex + maxStreak));
+
+        if (rightA.length !== 0 || rightB.length !== 0) {
+            output.push(...calculatePlaceholder(rightA, rightB));
+        }
+
+        return output;
+    };
+
+    // Merge A (missing placeholders) into B (can contain placeholders but does not have to)
+    const mergeMessages = (a: string, b: string, storage: Record<string, unknown>) => {
+        const placeholder = calculatePlaceholder(
+            normalizeMessage(a).split(' '),
+            normalizeMessage(b).split(' '),
+        ).join(' ');
+
+        if (placeholder === '_') {
+            return undefined;
+        }
+
+        interface HasCount {
+            count: number;
+        }
+
+        const count = (storage[a] as HasCount).count + (storage[b] as HasCount).count;
+
+        delete storage[a];
+        delete storage[b];
+
+        storage[placeholder] = {
+            count,
+        };
+
+        return placeholder;
+    };
+
+    const getErrorMessageGroup = (error: ErrnoException, storage: Record<string, unknown>) => {
+        let { message } = error;
+
+        if (!(message in storage)) {
+            storage[message] = {
+                count: 0,
+            };
+
+            for (const existingMessage of Object.keys(storage)) {
+                if (isSimilarMessage(message, existingMessage)) {
+                    const newMessage = mergeMessages(message, existingMessage, storage);
+                    if (newMessage) {
+                        message = newMessage;
+                        break;
+                    }
+                }
+            }
+        }
+
+        return storage[message] as Record<string, unknown>;
+    };
+
+    const increaseCount = (group: { count: number }) => {
+        if (!('count' in group)) {
+            // In case users don't want to display error message
+            group.count = 0;
+        }
+
+        group.count++;
+    };
+
+    let total = 0;
```

Comment:
Won't this be shared across all error handlers? Is that intended?

Reply:
How so? This is a local variable, not a global one.

Reply:
Oh wait, woops, misread the context and how it works 😅 . Ignore meee